Tests/test_file_avif.py

```diff
@@ -0,0 +1,537 @@
+            frames[0].copy().save(temp_file1, save_all=True, append_images=frames[1:])
+            check(temp_file1)
+
+            # Tests appending using a generator
+            def imGenerator(ims):
+                yield from ims
+
+            temp_file2 = str(tmp_path / "temp_generator.avif")
+            frames[0].copy().save(
+                temp_file2,
+                save_all=True,
+                append_images=imGenerator(frames[1:]),
+            )
+            check(temp_file2)
+
+    def test_timestamp_and_duration(self, tmp_path):
+        """
+        Try passing a list of durations, and make sure the encoded
+        timestamps and durations are correct.
+        """
+
+        durations = [1, 10, 20, 30, 40]
+        temp_file = str(tmp_path / "temp.avif")
+        with self.star_frames() as frames:
+            frames[0].save(
+                temp_file,
+                save_all=True,
+                append_images=(frames[1:] + [frames[0]]),
+                duration=durations,
+            )
+
+        with Image.open(temp_file) as im:
+            assert im.n_frames == 5
+            assert im.is_animated
+
+            # Check that timestamps and durations match original values specified
+            ts = 0
+            for frame in range(im.n_frames):
+                im.seek(frame)
+                im.load()
+                assert im.info["duration"] == durations[frame]
+                assert im.info["timestamp"] == ts
+                ts += durations[frame]
+
+    def test_seeking(self, tmp_path):
+        """
+        Create an animated AVIF file, and then try seeking through frames in
+        reverse-order, verifying the timestamps and durations are correct.
+        """
+
+        dur = 33
+        temp_file = str(tmp_path / "temp.avif")
+        with self.star_frames() as frames:
+            frames[0].save(
+                temp_file,
+                save_all=True,
+                append_images=(frames[1:] + [frames[0]]),
+                duration=dur,
+            )
+
+        with Image.open(temp_file) as im:
+            assert im.n_frames == 5
+            assert im.is_animated
+
+            # Traverse frames in reverse, checking timestamps and durations
+            ts = dur * (im.n_frames - 1)
+            for frame in reversed(range(im.n_frames)):
+                im.seek(frame)
+                im.load()
+                assert im.info["duration"] == dur
+                assert im.info["timestamp"] == ts
+                ts -= dur
+
+    def test_seek_errors(self):
+        with Image.open("Tests/images/avif/star.avifs") as im:
+            with pytest.raises(EOFError):
+                im.seek(-1)
+
+            with pytest.raises(EOFError):
+                im.seek(42)
+
+
+if hasattr(os, "sched_getaffinity"):
+    MAX_THREADS = len(os.sched_getaffinity(0))
+else:
+    MAX_THREADS = os.cpu_count()
+
+
+@skip_unless_feature("avif")
+class TestAvifLeaks(PillowLeakTestCase):
```

Comment:
I'd prefer not iterating a leak test in the standard test suite, as that can be expensive from a time POV. It's ok for the initial cut, but I'd rather not have it long term.